diff --git a/configure.ac b/configure.ac
index fc706c2..05547e3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -17,16 +17,16 @@ AC_HEADER_STDC
 CC_NOUNDEFINED
 
 PKG_CHECK_MODULES(ALSA, alsa >= 1.1.6)
-case $host_os in
-netbsd* | freebsd* | dragonfly* | openbsd*)
-  AC_CHECK_LIB(asound, snd_pcm_ioplug_create,,
-	     AC_ERROR([*** libasound has no external plugin SDK]))
-  ;;
-*)
-  AC_CHECK_LIB(asound, snd_pcm_ioplug_create,,
-	     AC_ERROR([*** libasound has no external plugin SDK]), -ldl)
-  ;;
-esac
+#case $host_os in
+#netbsd* | freebsd* | dragonfly* | openbsd*)
+#  AC_CHECK_LIB(asound, snd_pcm_ioplug_create,,
+#	     AC_ERROR([*** libasound has no external plugin SDK]))
+#  ;;
+#*)
+#  AC_CHECK_LIB(asound, snd_pcm_ioplug_create,,
+#	     AC_ERROR([*** libasound has no external plugin SDK]), -ldl)
+#  ;;
+#esac
 
 AC_ARG_ENABLE([oss],
       AS_HELP_STRING([--disable-oss], [Disable building of OSS plugin]))
