diff --git a/modules/pam_namespace/pam_namespace.c b/modules/pam_namespace/pam_namespace.c
index f72d6718..3a6597af 100644
--- a/modules/pam_namespace/pam_namespace.c
+++ b/modules/pam_namespace/pam_namespace.c
@@ -38,6 +38,7 @@
 #include "pam_inline.h"
 #include "pam_namespace.h"
 #include "argv_parse.h"
+#include <stdint.h>
 
 /* --- evaluating all files in VENDORDIR/security/namespace.d and /etc/security/namespace.d --- */
 static const char *base_name(const char *path)
